formulaic/transforms/basis_spline.py

```diff
@@ -136,12 +150,19 @@ def basis_spline(  # pylint: disable=dangerous-default-value  # always replaced
                 raise ValueError(
                     f"Invalid value for `df`. `df` must be greater than {degree + (1 if include_intercept else 0)} [`degree` (+ 1 if `include_intercept` is `True`)]."
                 )
-            knots = list(
-                numpy.quantile(x, numpy.linspace(0, 1, nknots + 2))[1:-1].ravel()
+            if knots_x.shape[0] < nknots:
```

Comment:
I think there was a comment that pointed to a different line. This could stay or go.  Could go to a minimal check that there is at least 1 data point here.